devops/jobs/ConfigurationWatcher.groovy

```diff
@@ -0,0 +1,154 @@
+    EDX_REPO_ROOT: the URL of the edX GitHub organization (REQUIRED)
+    CONFIG_JSON_FILE_CREDENTIAL_ID: ID of the credentials set up in Jenkins to log into DockerHub; defaults to docker-config-json; this refers to the docker-config-json expected credential described below
+    FOLDER_NAME: the name of the folder in which the downstream per-app jobs will live; defaults to DockerCI
+    ACCESS_CONTROL: list of github users or groups who will have access to the jobs
+    Expected credentials - these will normally be set up on the Folder.
+    docker-config-json: a file credential that gives each job in the folder permission to push to DockerHub
+*/
+
+package devops.jobs
+
+import static org.edx.jenkins.dsl.YAMLHelpers.parseYaml
+import org.yaml.snakeyaml.error.YAMLException
+
+import static org.edx.jenkins.dsl.DevopsConstants.common_read_permissions
+import static org.edx.jenkins.dsl.Constants.common_logrotator
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+
+class ConfigurationWatcher {
+    public static def job = { dslFactory, extraVars ->
+        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/configuration-watcher") {
+
+            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+            assert extraVars.containsKey('EDX_REPO_ROOT') : "Please define EDX_REPO_ROOT"
+            assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+
+            def config_branch = 'master'
+
+            // to space separated list as string
+            def apps = extraVars.get("APPS_TO_CONFIG").keySet().join(",").replaceAll(',', " ")
+
+            // inject APPS as an environment variable to be used by trigger-builds.sh
+            environmentVariables {
+                env('APPS', apps)
+            }
+
+            wrappers common_wrappers
+            // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+            wrappers {
+                credentialsBinding {
+                    file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                }
+            }
+            logRotator common_logrotator
+
+            def access_control = extraVars.get('ACCESS_CONTROL',[])
+            access_control.each { acl ->
+                common_read_permissions.each { perm ->
+                    authorization {
+                        permission(perm,acl)
+                    }
+                }
+            }
+
+            // for GitHub Webhooks
+            properties {
+                githubProjectUrl(extraVars.get("EDX_REPO_ROOT") +  "configuration/")
+            }
+
+            scm {
+                // check out edx/configuration repository from GitHub
+                git {
+                    remote {
+                        url(extraVars.get("CONFIGURATION_REPO_URL"))
```

Comment:
You shouldn't need the .git on 95, github'll clone with or without just fine, but you would need the / at 88.

I think we should probably switch to using CONFIGURATION_REPO_URL in both places and documenting that it has to be https for the githubproject part use case.  That'll help anyone who uses this with an ssh repo later.

It's a little "pick your failure case" since either way can cause pain to change.  I'll leave final call to your judgement.